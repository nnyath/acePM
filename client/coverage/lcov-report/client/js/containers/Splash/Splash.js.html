<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for client/js/containers/Splash/Splash.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">client/js/containers/Splash</a> Splash.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">93.94% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>31/33</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>10/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>9/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">93.94% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>31/33</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">15x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">29x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">13x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">62x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">62x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import {
  Text,
  View,
  Button,
  Image
} from 'react-native';
import { connect } from 'react-redux';
&nbsp;
import update from 'immutability-helper'
import styles from './styles'
import {_wait} from '../../util/tests/testutil'
import {initLoad} from './actions'
import consts from '../../util/consts'
&nbsp;
@connect(store =&gt; {
  return {
    init : store.main.init
  }
})
export default class Splash extends Component {
&nbsp;
  constructor() {
    super()
    this.state = {
      info:'Loading',
      timeout: 1000,
      ellipses : {
        rate : 500,
        handler : null,
        display: ''
      }
    }
&nbsp;
    this.startEllipses = this.startEllipses.bind(this)
    this.stopEllipses = this.stopEllipses.bind(this)
    this.simulateLoading = this.simulateLoading.bind(this)
    this.iterateMsg = this.iterateMsg.bind(this)
  }
&nbsp;
  componentDidMount(){
      this.simulateLoading(consts.UI.LOADING_MESSAGES, 10, this.state.timeout)
  }
&nbsp;
  async simulateLoading(loadingMsgs,displayCount, timeout){
        
    this.startEllipses()
&nbsp;
    try{
      await this.iterateMsg(loadingMsgs,displayCount, timeout)
    }catch(err){
<span class="cstat-no" title="statement not covered" >      console.warn(err)</span>
    }
    
    this.stopEllipses()
    this.props.dispatch(initLoad())
  }
&nbsp;
  startEllipses(){
    return this.setState(
      update(this.state, { ellipses: { handler: { $set: setInterval(()=&gt;{
        this.setState(update(this.state, {ellipses:{display:{$set: this.state.ellipses.display.length &gt; 2 ? '.' : this.state.ellipses.display+'.' }}}))
      },this.state.ellipses.rate) } } })
    )
  }
&nbsp;
  async stopEllipses(){
    clearInterval(this.state.ellipses.handler)
    return await this.setState(update(this.state, { ellipses: { display: { $set: '' }, handler: { $set: null } } }))
  }
&nbsp;
  /* Unpure (setState side-effect) recursive helper function to wait and iterate through loading messages*/
  iterateMsg = async (msgs, count, timeout) =&gt; {
    
    if(!Array.isArray(msgs))
      throw new Error('Invalid loading messages')
    
    if(count&lt;=0)
      return msgs === [] ? <span class="branch-0 cbranch-no" title="branch not covered" >[] </span>: msgs
    
      
    if(count&gt;msgs.length){
      console.warn('Cannot display more messages than what exists, defaulting to messages.length')
      return await this.iterateMsg(msgs,msgs.length,timeout)
    }
&nbsp;
    <span class="missing-if-branch" title="if path not taken" >I</span>if(msgs.length==0){
<span class="cstat-no" title="statement not covered" >      throw new Error('Invalid loading messages')</span>
    }
&nbsp;
    await _wait(timeout)
    const randInd = Math.floor(Math.random()*msgs.length)
    await this.setState(update(this.state, {info:{$set:msgs[randInd]}}))
    return await this.iterateMsg(update(msgs,{$splice:[[randInd,1]]}),count-1,timeout)
  }
&nbsp;
  render() {
&nbsp;
    const {info, ellipses} = this.state
&nbsp;
    return (
      &lt;View style={styles.container}&gt;
        &lt;View style={styles.img}&gt;
          &lt;Image style={styles.logo}
            source={require('../../../media/img/acePM_redux.png')}&gt;
          &lt;/Image&gt;
          &lt;Text style={styles.info}&gt;{info}{ellipses.display}&lt;/Text&gt;
        &lt;/View&gt;
      &lt;/View&gt;
    );
  }
}
&nbsp;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jul 04 2017 02:07:51 GMT-0700 (PDT)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
