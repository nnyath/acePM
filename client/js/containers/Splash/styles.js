import { StyleSheet } from 'react-native';

export default StyleSheet.create({
    container: {
        flex:1,
        flexDirection:'column',
        justifyContent:'center',
        alignItems:'center'
    },
    img:{
        width:300,
        height:300,
    },
    info: {
        fontSize: 20,
        textAlign: 'center',
        margin: 10,
    },
    logo : {
        flex:1,
        width:null,
        height:null,
        resizeMode : 'center',
    }
})