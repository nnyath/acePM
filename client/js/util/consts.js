export default {
    UI : {
        RECORD_BUTTON : {
            ID: 'recordButton',
            START: 'Start Recording',
            STOP : 'Stop Recording'
        },
        LOADING_MESSAGES : [
            'Powering Streams',
            'Crossing Overs',
            'Dividing 16th Notes',
            'Modifying Speed',
            'Cooling Freezes',
            'Capturing Air',
            'Embracing Holds'
        ]
    },
    UTIL : {
        WAIT_TIME : 1
    },
    REDUX : {
        INIT_LOADING : 'INIT_LOADING',
        INIT_LOADING_PENDING : 'INIT_LOADING_PENDING',
        INIT_LOADING_FULFILLED : 'INIT_LOADING_FULFILLED',
        INIT_LOADING_REJECTED : 'INIT_LOADING_REJECTED'
    }
}