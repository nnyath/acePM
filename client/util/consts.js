export default {
    UI : {
        RECORD_BUTTON : {
            ID: 'recordButton',
            START: 'Start Recording',
            STOP : 'Stop Recording'
        }
    },
    UTIL : {
        WAIT_TIME : 1
    }
}