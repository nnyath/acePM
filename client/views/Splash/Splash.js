import React, { Component } from 'react';
import {
  Text,
  View,
  Button,
  Image
} from 'react-native';

import update from 'immutability-helper'
import styles from './styles'
import {_wait} from '../../util/tests/testutil'

export default class Splash extends Component {

  constructor() {
    super()
    this.state = {
      info:'Loading',
      ellipses : {
        rate : 500,
        handler : null,
        display: ''
      }
    }
    this.startEllipses = this.startEllipses.bind(this)
    this.stopEllipses = this.stopEllipses.bind(this)
  }

  async componentDidMount(){
    this.startEllipses()
    await _wait(5000)
    await this.setState(update(this.state, {info:{$set:'Connecting to Server'}}))
    await _wait(5000)
    await this.setState(update(this.state, {info:{$set:'Calibrating 16th Notes'}}))
    await _wait(5000)
    await this.setState(update(this.state, {info:{$set:'Mapping Crossovers'}}))
    await _wait(20000)
    this.stopEllipses()
    
  }

  startEllipses(){
    this.setState(
      update(this.state, { ellipses: { handler: { $set: setInterval(()=>{
        this.setState(update(this.state, {ellipses:{display:{$set: this.state.ellipses.display.length > 2 ? '' : this.state.ellipses.display+'.' }}}))
      },this.state.ellipses.rate) } } })
    )
  }

  stopEllipses(){
    clearInterval(this.state.ellipses.handler)
    this.setState(update(this.state, { ellipses: { display: { $set: '' }, handler: { $set: null } } }))
  }

  render() {

    const {info, ellipses} = this.state

    return (
      <View style={styles.container}>
        <View style={styles.img}>
          <Image style={styles.logo}
            source={require('../../media/img/acePM_redux.png')}
          >
          </Image>
          <Text style={styles.info}>{info}{ellipses.display}</Text>
        </View>
      </View>
    );
  }
}

