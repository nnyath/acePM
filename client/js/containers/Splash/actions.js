import consts from '../../util/consts'
import {_wait} from '../../util/tests/testutil'

export const initLoad = () => ({
    type: consts.REDUX.INIT_LOADING,
    payload: new Promise((resolve,reject) => {
        _wait(3000).then(res => resolve(true))
    }).catch(err => reject(err))
})