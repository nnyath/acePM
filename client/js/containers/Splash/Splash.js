import React, { Component } from 'react';
import {
  Text,
  View,
  Button,
  Image
} from 'react-native';
import { connect } from 'react-redux';

import update from 'immutability-helper'
import styles from './styles'
import {_wait} from '../../util/tests/testutil'
import {initLoad} from './actions'
import consts from '../../util/consts'

@connect(store => {
  return {
    init : store.main.init
  }
})
export default class Splash extends Component {

  constructor() {
    super()
    this.state = {
      info:'Loading',
      ellipses : {
        rate : 500,
        handler : null,
        display: ''
      }
    }

    this.startEllipses = this.startEllipses.bind(this)
    this.stopEllipses = this.stopEllipses.bind(this)

    this.simulateLoading = this.simulateLoading.bind(this)
  }

  componentDidMount(){
      this.simulateLoading(3)
  }

  async simulateLoading(displayCount){
    /* Pure recursive helper function to wait and iterate through loading messages*/
    iterateMsg = async (msgs, count) => {
      if(count<=0)
        return

      await _wait(5000)
      const randInd = Math.floor(Math.random()*msgs.length)
      await this.setState(update(this.state, {info:{$set:msgs[randInd]}}))
      await iterateMsg(update(msgs,{$splice:[[randInd,1]]}),count-1)
    }
    
    this.startEllipses()
    await iterateMsg(consts.UI.LOADING_MESSAGES,displayCount)
    this.stopEllipses()
    
    this.props.dispatch(initLoad())
  }

  startEllipses(){
    this.setState(
      update(this.state, { ellipses: { handler: { $set: setInterval(()=>{
        this.setState(update(this.state, {ellipses:{display:{$set: this.state.ellipses.display.length > 2 ? '' : this.state.ellipses.display+'.' }}}))
      },this.state.ellipses.rate) } } })
    )
  }

  stopEllipses(){
    clearInterval(this.state.ellipses.handler)
    this.setState(update(this.state, { ellipses: { display: { $set: '' }, handler: { $set: null } } }))
  }

  render() {

    const {info, ellipses} = this.state

    return (
      <View style={styles.container}>
        <View style={styles.img}>
          <Image style={styles.logo}
            source={require('../../../media/img/acePM_redux.png')}>
          </Image>
          <Text style={styles.info}>{info}{ellipses.display}</Text>
        </View>
      </View>
    );
  }
}

