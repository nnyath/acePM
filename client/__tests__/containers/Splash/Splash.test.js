import React from 'react'
import { shallow } from 'enzyme'
import update from 'immutability-helper'
import Sinon from 'sinon'

import configureStore from 'redux-mock-store'

import {_setState,_state,_wait} from '../../../js/util/tests/testutil'
import consts from '../../../js/util/consts'

import Splash from '../../../js/containers/Splash/Splash'


describe('init', ()=>{

    const initialState = { main : {init: false }} 
    let store, wrapper

    beforeEach(()=>{
        store = configureStore()(initialState)
        wrapper = shallow(<Splash/>, {context:{store:store}})
    })

    afterEach(()=>{
    })

    it('renders correctly', () => {
        expect(wrapper.dive()).toMatchSnapshot()
    })

})



