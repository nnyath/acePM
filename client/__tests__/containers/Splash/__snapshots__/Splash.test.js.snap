// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "alignItems": "center",
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
          }
    }
>
    <View
        style={
            Object {
                "height": 300,
                "width": 300,
              }
        }
    >
        <Image
            source={1}
            style={
                Object {
                    "flex": 1,
                    "height": null,
                    "resizeMode": "center",
                    "width": null,
                  }
            }
        />
        <Text
            accessible={true}
            allowFontScaling={true}
            disabled={false}
            ellipsizeMode="tail"
            style={
                Object {
                    "fontSize": 20,
                    "margin": 10,
                    "textAlign": "center",
                  }
            }
        >
            Loading
            
        </Text>
    </View>
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                }
      }
>
      <View
            style={
                  Object {
                        "height": 300,
                        "width": 300,
                      }
            }
      >
            <Image
                  source={1}
                  style={
                        Object {
                              "flex": 1,
                              "height": null,
                              "resizeMode": "center",
                              "width": null,
                            }
                  }
            />
            <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                        Object {
                              "fontSize": 20,
                              "margin": 10,
                              "textAlign": "center",
                            }
                  }
            >
                  Loading
                  
            </Text>
      </View>
</View>,
  ],
  "options": Object {
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
  "renderer": ReactShallowRenderer {
    "_context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "_element": <Splash
      dispatch={[Function]}
      init={false}
/>,
    "_instance": Splash {
      "context": Object {
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "props": Object {
        "dispatch": [Function],
        "init": false,
      },
      "refs": Object {},
      "simulateLoading": [Function],
      "startEllipses": [Function],
      "state": Object {
        "ellipses": Object {
          "display": "",
          "handler": null,
          "rate": 500,
        },
        "info": "Loading",
      },
      "stopEllipses": [Function],
      "updater": Updater {
        "_renderer": [Circular],
      },
    },
    "_newState": null,
    "_rendered": <View
      style={
            Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                }
      }
>
      <View
            style={
                  Object {
                        "height": 300,
                        "width": 300,
                      }
            }
      >
            <Image
                  source={1}
                  style={
                        Object {
                              "flex": 1,
                              "height": null,
                              "resizeMode": "center",
                              "width": null,
                            }
                  }
            />
            <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                        Object {
                              "fontSize": 20,
                              "margin": 10,
                              "textAlign": "center",
                            }
                  }
            >
                  Loading
                  
            </Text>
      </View>
</View>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Splash
    dispatch={[Function]}
    init={false}
/>,
}
`;
